containenv/containenv/cli/commands/test/test_init.py

```diff
@@ -0,0 +1,238 @@
+    EXPECTED_ERROR = ('init can only be called once per project, but the {}\n'
+                      'project already has a ".containenv" directory.'
+                      ''.format(testdirectory))
+    with pytest.raises(ProjectAlreadyInitialized) as PAIEIO:
+        init_command.check_initialization_state()
+    assert PAIEIO.value.args[0] == EXPECTED_ERROR
+
+
+def test__catalog_dependencies_none_registered(testdirskeleton):
+    '''none of the nodes are registered as containenv dependencies'''
+    testdirectory, CONTAINENVPATH, SUBDIRPATH, init_command = testdirskeleton
+    EXPECTED_UNREGISTERED = {SUBDIRPATH, CONTAINENVPATH}
+    fnames = [os.path.join(testdirectory, n) for n in [
+                os.path.join(SUBDIRPATH, 'foo.txt'),
+                'setup.cfg',
+                'setup']]
+    for fn in fnames:
+        open(fn, 'w').write('TESTTEXT')
+        EXPECTED_UNREGISTERED.add(fn)
+    with pytest.raises(ProjectContainsNoRegisteredNodes) as PCNRNEIO:
+        init_command._catalog_dependencies()
+    assert init_command.unregistered_nodes == EXPECTED_UNREGISTERED
+    assert PCNRNEIO.value.args[0] == \
+        ('None of the nodes (files and directories) in this project are'
+         ' registered with containenv.')
+
+
+@pytest.fixture
+def testdirgeneric(testdirskeleton):
+    infiles = (os.path.join('SUB', 'foo.go'),
+               'Makefile.build',
+               'setup.sh',
+               'py.foo',
+               os.path.join('.git', 'didntgetregistered.py'))
+
+    def _make_flesh(list_of_files=infiles):
+        testdirectory, CONTAINENVPATH, SUBDIRPATH, init_command =\
+            testdirskeleton
+        gitdirpath = os.path.join(testdirectory, '.git')
+        os.mkdir(gitdirpath)
+        fnames = [os.path.join(testdirectory, n) for n in list_of_files]
+        for fn in fnames:
+            open(fn, 'w').write('TESTTEXT')
+        return (testdirectory,
+                CONTAINENVPATH,
+                SUBDIRPATH,
+                init_command,
+                gitdirpath,
+                fnames)
+    return _make_flesh
+
+
+def test__catalog_dependencies_bash_go_git_makefile(testdirgeneric):
+    (testdirectory,
+     CONTAINENVPATH,
+     SUBDIRPATH,
+     init_command,
+     gitdirpath,
+     fnames) = testdirgeneric()
+    init_command._catalog_dependencies()
+    assert init_command.registered_dependency_catalog.pop('golang') ==\
+        set([fnames[0]])
+    assert init_command.registered_dependency_catalog.pop('make') ==\
+        set([fnames[1]])
+    assert init_command.registered_dependency_catalog.pop('bash') ==\
+        set([fnames[2]])
+    assert init_command.registered_dependency_catalog.pop('git') ==\
+        set([gitdirpath])
+    assert not init_command.registered_dependency_catalog
+    assert init_command.unregistered_nodes ==\
+        {fnames[3], SUBDIRPATH, CONTAINENVPATH}
+
+
+BASH_GOT_GIT_MAKE_TMPL = 'FROM        ubuntu:latest\n' +\
```

Comment:
Use `r"""blah blah blah"""` for this so you don't have all the line wrapping and escaping.